src/main/java/com/thealgorithms/sorts/MergeSortNoExtraSpace.java

```diff
@@ -1,73 +1,83 @@
 package com.thealgorithms.sorts;
 
 import java.util.Arrays;
-import java.util.Scanner;
 
-/*This code implements the mergeSort algorithm without extra space
-For understanding about mergesort visit :https://www.geeksforgeeks.org/merge-sort/
+/**
+ * Implementation of Merge Sort without using extra space for merging.
+ * This implementation performs in-place merging to sort the array of integers.
  */
 public final class MergeSortNoExtraSpace {
     private MergeSortNoExtraSpace() {
     }
 
-    public static void callMergeSort(int[] a, int n) {
-        int maxele = Arrays.stream(a).max().getAsInt() + 1;
-        mergeSort(a, 0, n - 1, maxele);
+    /**
+     * Sorts the array using in-place merge sort algorithm.
+     *
+     * @param array the array to be sorted
+     * @return the sorted array
+     */
+    public static int[] sort(int[] array) {
```

Comment:
This implementation does not handle negative inputs: is that right? If yes, please throw `IllegalArgumentException` in such situation.

Reply:
Yes, you are right. Will add this check and test for it.